test/apps/iodemo/io_demo.cc

```diff
@@ -172,21 +250,75 @@ class IoDemoRandom {
         }
     }
 
-    static inline void fill(unsigned &seed, void *buffer, size_t size) {
+    static void *get_host_fill_buffer(void *buffer, size_t size,
+                                      ucs_memory_type_t memory_type)
+    {
+#ifdef HAVE_CUDA
+        void *buffer_p = buffer;
+        static std::vector<uint8_t> _buffer;
+
+        if (memory_type == UCS_MEMORY_TYPE_CUDA) {
+            _buffer.resize(size);
+            buffer_p = _buffer.data();
```

Comment:
can we just "return _buffer.data()"? and avoid the local var buffer_p
and in line 265: return buffer